pygmt/helpers/decorators.py

```diff
@@ -338,9 +357,15 @@ def new_module(*args, **kwargs):
                     value = kwargs[arg]
                     issequence = fmt in separators
                     if issequence and is_nonstr_iter(value):
-                        kwargs[arg] = separators[fmt].join(
-                            "{}".format(item) for item in value
-                        )
+                        for index, item in enumerate(value):
+                            try:
+                                assert " " not in str(item)
+                            except AssertionError:
+                                # convert datetime-like items to string format
+                                value[index] = np.datetime_as_string(
+                                    np.asarray(item, dtype=np.datetime64)
```

Comment:
We use `np.datetime_as_string(array_to_datetime(vector))` to convert datetime arrays to string arrays. 
https://github.com/GenericMappingTools/pygmt/blob/2ddbb0e1977af85bf6a6ac63d7cbac9da3d08f05/pygmt/clib/session.py#L778-L782
The `array_to_datetime` function uses `pd.to_datetime(array)`
https://github.com/GenericMappingTools/pygmt/blob/2ddbb0e1977af85bf6a6ac63d7cbac9da3d08f05/pygmt/clib/conversion.py#L324

Any specific reason to use `np.asarray` here?

Reply:
Main reason is to handle `xarray.DataArray` inputs. The `array_to_datetime`/`pd.to_datetime` function is meant to handle arrays/lists, but not scalar values. I.e. this doesn't work:

```python
import pandas as pd
import xarray as xr

pd.to_datetime(xr.DataArray(data=np.datetime64("2005-01-01")))
# TypeError: len() of unsized object
```

but this does:

```python
import numpy as np
import xarray as xr

np.asarray(xr.DataArray(data=np.datetime64("2005-01-01")))
# array('2005-01-01T00:00:00.000000000', dtype='datetime64[ns]')
```